electra/the fun part/utilities/utils.c

```diff
@@ -15,23 +15,23 @@
 
 // Thanks @nitotv
 // https://ghostbin.com/paste/rmvp5
-//int run(const char *cmd) {
-//    pid_t pid;
-//    int rv;
-//
-//    char *environ[] = {
-//        "BOOTSTRAP_PREFIX=/"BOOTSTRAP_PREFIX"",
-//        "PATH=/"BOOTSTRAP_PREFIX"/usr/local/bin:/"BOOTSTRAP_PREFIX"/usr/sbin:/"BOOTSTRAP_PREFIX"/usr/bin:/"BOOTSTRAP_PREFIX"/sbin:/"BOOTSTRAP_PREFIX"/bin:/bin:/usr/bin:/sbin:/usr/sbin"
-//    };
-//
-//    char *argv[] = {"sh", "-c", (char*)cmd, NULL};
-//    rv = posix_spawn(&pid, "/bin/sh", NULL, NULL, argv, environ);
-//
-//    if (rv == 0) {
-//        if (waitpid(pid, &rv, 0) == -1)
-//            perror("waitpid");
-//    }
-//
-//    return rv;
-//}
+int run(const char *cmd) {
+    pid_t pid;
+    int rv;
+
+    char *environ[] = {
+        "BOOTSTRAP_PREFIX=/"BOOTSTRAP_PREFIX"",
+        "PATH=/"BOOTSTRAP_PREFIX"/usr/local/bin:/"BOOTSTRAP_PREFIX"/usr/sbin:/"BOOTSTRAP_PREFIX"/usr/bin:/"BOOTSTRAP_PREFIX"/sbin:/"BOOTSTRAP_PREFIX"/bin:/bin:/usr/bin:/sbin:/usr/sbin"
+    };
+
+    char *argv[] = {"sh", "-c", (char*)cmd, NULL};
+    rv = posix_spawn(&pid, "/bin/sh", NULL, NULL, argv, environ);
```

Comment:
do we really have /bin/sh?

Reply:
No, we don't

Reply:
It’s meant to be /bootstrap, nice find

Reply:
you don't call that function before gnubinpack is unpacked, right?